crates/client/src/config.rs

```diff
@@ -0,0 +1,101 @@
+extern crate lazy_static;
+
+use anyhow::Context;
+use fehler::throw;
+use serde::Deserialize;
+use std::{env, fs, io, path::PathBuf};
+use thiserror::Error;
+
+pub const CARGO_TOML: &str = "Cargo.toml";
+pub const TRDELNIK_TOML: &str = "Trdelnik.toml";
+pub const ANCHOR_TOML: &str = "Anchor.toml";
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("invalid workspace")]
+    BadWorkspace,
+    #[error("{0:?}")]
+    Anyhow(#[from] anyhow::Error),
+    #[error("{0:?}")]
+    Io(#[from] io::Error),
+    #[error("{0:?}")]
+    Toml(#[from] toml::de::Error),
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct Test {
+    pub validator_startup_timeout: u64,
+}
+
+#[derive(Default, Debug, Deserialize, Clone)]
+struct _Test {
+    #[serde(default)]
+    pub validator_startup_timeout: Option<u64>,
+}
+
+impl From<_Test> for Test {
+    fn from(_t: _Test) -> Self {
+        Self {
+            validator_startup_timeout: _t.validator_startup_timeout.unwrap_or(10_000),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct Config {
+    pub test: Test,
+}
+
+#[derive(Default, Debug, Deserialize, Clone)]
+struct _Config {
```

Comment:
I was quite confused by the naming of these structs. If you find a better names, it would be good. But it's not a blocker.